src/xmipp/libraries/reconstruction/angular_distance.cpp

```diff
@@ -465,7 +468,7 @@ void ProgAngularDistance::computeWeights()
 				ang1.clear();
 				ang2.clear();
     		}
-        anotherImageIn2=iter2.hasNext();
+        anotherImageIn2 = iter2 != DF2sorted.ids().end();
```

Comment:
consider adding `hasNext()` into API

Reply:
`it != obj.end()` is standard method of checking end. TODO discuss.

Reply:
Add `hasNext` to all iterators.